### PR TITLE
Fix HumanEntity#drop not updating the client inventory

### LINKED ISSUE #6760: Player::dropItem not work like expected
### Expected behavior

When use the method `player.dropItem(true)` the player drop the item (all stack) in hand.

### Observed/Actual behavior

When use the method the item drop but still show in hotbar, but if you click the item in hotbar then disapear.

### Steps/models to reproduce

Using player.dropItem in any cases.

### Plugin and Datapack List

Plugins: ConterCraft, CoreProtect, FastAsyncWorldEdit (WorldEdit), HackedCore*, HackedServer*, LibsDisguises, LuckPerms, PlugMan, ProtocolLib, SentryIntegration, spark

DataPacks: [vanilla (built-in)], [file/bukkit (world)], [file/VoidDimension-v1.2.zip (world)], [file/InfectionAdvancements-v.1.zip (world)], [file/FarmingLandDimension-v1.0.zip (world)], [file/BunnyBurrowDimension-v1.0.zip (world)]

### Paper version

Paper version git-Paper-325 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: 4a0d935)

### Other

This working fine in 1.16.5, in the last version need use updateInventory for avoid glitch.

### PR DESCRIPTION
**Pull request by Machine-Maker**
Fixes #6760

### CODE REVIEW

File: patches/server/0826-Fix-HumanEntity-drop-not-updating-the-client-inv.patch
Comment (kennytv) [inline patches/server/0826-Fix-HumanEntity-drop-not-updating-the-client-inv.patch]: This results in unnecessary updates in case of even cancellation (and for any call that isn't from the HumanEntity#dropItem method), you should move it up to the actual cb method

**Review by kennytv — DISMISSED** [review]

**Comment by Machine-Maker** [comment]
Rebased for 1.18.1

**Review by Owen1212055 — APPROVED** [review]
Fixes the issue, tested...

**Review by NoahvdAa — APPROVED** [review]

**Review by lynxplay — COMMENTED** [review]
The AT should be moved into the patch before merge, presumably our new maintainer can do that while merging tho :)

File: patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch
Comment (lynxplay) [inline patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch]: What is preventing us from technically returning early here ?
`success` should be false only if no item was dropped whatsoever, so we could skip some logic here unless I am missing something.

Comment (Machine-Maker) [reply patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch]: That's the purpose of the PR. To fix the desync. The server can cancel the drop when the client expects it to drop, so we need to resync.

**Comment by Machine-Maker** [comment]
Rebased for 1.19.3

File: patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch
Comment (lynxplay) [reply patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch]: Was it ? At least in my testing the issue was that the client specifically does *not* expect an item drop and hence
does not update the inventory.

> When use the method the item drop but still show in hotbar, but if you click the item in hotbar then disapear.
The issue seems to match that.

Comment (Machine-Maker) [reply patches/server/0878-Fix-HumanEntity-drop-not-updating-the-client-inv.patch]: Yeah, that sounds right. I guess I totally forgot what this was about. The cancellation logic for PlayerDropItemEvent handles updating the inventory.

**Review by lynxplay — APPROVED** [review]

**Pull request merged**
